snuba/datasets/storages/sessions.py

```diff
@@ -111,9 +111,9 @@
             countIfState(session_id, status == 2) as sessions_crashed,
             countIfState(session_id, status == 3) as sessions_abnormal,
             uniqIfState(session_id, errors > 0) as sessions_errored,
-            uniqIfState(distinct_id, status == 2) as users_crashed,
-            uniqIfState(distinct_id, status == 3) as users_abnormal,
-            uniqIfState(distinct_id, errors > 0) as users_errored
+            uniqIfState(distinct_id, distinct_id != '{NIL_UUID}' and status == 2) as users_crashed,
```

Comment:
This change will be ok for new materialized views created from now on, but it does not take migrations of existing ones into account. Unfortunately the process is a bit involved and quite hard in prod.

For dev you will need to define a migrations like we were attempting here
https://github.com/getsentry/snuba/pull/702/files#diff-d4c38d96ce82126b38a8f17fc3b0942dR69-R90
even though. never committed, where the matview is detached and recreated.

For production and other. environments it will be more complex, you should sync with @JTCunning to perform the change there since it is a manual process and it has several limitation.